From a9383896995b40187442c53c9651f7e73a0c5285 Mon Sep 17 00:00:00 2001
From: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Date: Thu, 2 Mar 2023 13:53:48 +0200
Subject: [PATCH 46/54] Revert "s32g399ardb3: add ddr config fixup for revision
 f"

This reverts commit 84658bd5439a821cc60416440f469798f611c2e9.

The ADC support in ATF is deleted as it's no longer needed for
identifying the RDB revision to apply a DDR fixup.

Issue: ALB-9722
Upstream-Status: Pending 

Signed-off-by: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 .../s32/s32g/s32g3/s32g399ardb3/platform.mk   |   2 -
 .../s32g399ardb3/s32g399ardb3_ddr_fixup.c     | 133 ------------------
 plat/nxp/s32/s32g/s32g_bl2_el3.c              |  12 --
 3 files changed, 147 deletions(-)
 delete mode 100644 plat/nxp/s32/s32g/s32g3/s32g399ardb3/s32g399ardb3_ddr_fixup.c

diff --git a/plat/nxp/s32/s32g/s32g3/s32g399ardb3/platform.mk b/plat/nxp/s32/s32g/s32g3/s32g399ardb3/platform.mk
index 63a832828..ef94c81b5 100644
--- a/plat/nxp/s32/s32g/s32g3/s32g399ardb3/platform.mk
+++ b/plat/nxp/s32/s32g/s32g3/s32g399ardb3/platform.mk
@@ -10,6 +10,4 @@ S32_BOARD_PATH		:= ${PLAT_SOC_PATH}/s32g399ardb3
 
 PLAT_INCLUDES		+= -I${S32_BOARD_PATH}/include \
 
-PLAT_BL_COMMON_SOURCES  += ${S32_BOARD_PATH}/s32g399ardb3_ddr_fixup.c
-
 DTB_FILE_NAME		?= s32g399a-rdb3.dtb
diff --git a/plat/nxp/s32/s32g/s32g3/s32g399ardb3/s32g399ardb3_ddr_fixup.c b/plat/nxp/s32/s32g/s32g3/s32g399ardb3/s32g399ardb3_ddr_fixup.c
deleted file mode 100644
index 587480286..000000000
--- a/plat/nxp/s32/s32g/s32g3/s32g399ardb3/s32g399ardb3_ddr_fixup.c
+++ /dev/null
@@ -1,133 +0,0 @@
-// SPDX-License-Identifier: BSD-3-Clause
-/*
- * Copyright 2022 NXP
- */
-#include <stdint.h>
-#include <common/debug.h>
-#include <common/fdt_wrappers.h>
-#include <drivers/nxp/s32/adc/s32_adc.h>
-#include "ddr_init.h"
-#include <lib/libc/errno.h>
-#include <libfdt.h>
-#include "s32_dt.h"
-
-#define SARADC0_CH5		5
-#define SARADC0_TOLERANCE	200
-
-#define UMCTL2_RFSHCTL0		0x403C0050
-#define UMCTL2_RFSHCTL0_RDB3F	0x80210010U
-
-static const struct {
-	const char *rev;
-	const char *desc;
-	uint32_t lower;
-	uint32_t upper;
-} rdb_revisions[] = {
-	{ /* 0V */
-		.rev = "",
-		.lower = 0,
-		.upper = 400,
-	},
-	{ /* 0.8v */
-		.rev = "C",
-		.lower = 1820 - SARADC0_TOLERANCE,
-		.upper = 1820 + SARADC0_TOLERANCE,
-	},
-	{ /* 1.0v */
-		.rev = "D",
-		.lower = 2275 - SARADC0_TOLERANCE,
-		.upper = 2275 + SARADC0_TOLERANCE,
-	},
-	{ /* 1.2v */
-		.rev = "E",
-		.lower = 2730 - SARADC0_TOLERANCE,
-		.upper = 2730 + SARADC0_TOLERANCE,
-	},
-	{ /* 1.4v */
-		.rev = "F",
-		.lower = 3185 - SARADC0_TOLERANCE,
-		.upper = 3185 + SARADC0_TOLERANCE,
-	},
-};
-
-static int find_rdb_rev(uint32_t adc_value)
-{
-	int i;
-
-	for (i = 0; i < ARRAY_SIZE(rdb_revisions); i++) {
-		if (rdb_revisions[i].lower <= adc_value &&
-		    adc_value <= rdb_revisions[i].upper)
-			return i;
-	}
-
-	return -EINVAL;
-}
-
-static int ddr_fixup_rdb_revf(void)
-{
-	size_t i;
-
-	for (i = 0; i < ddrc_cfg_size; i++)
-		if (ddrc_cfg[i].addr == UMCTL2_RFSHCTL0) {
-			ddrc_cfg[i].data = UMCTL2_RFSHCTL0_RDB3F;
-			break;
-		}
-
-	return 0;
-}
-
-static int dt_init_adc(struct s32_adc *adc_driver)
-{
-	void *fdt = NULL;
-	int adc_node;
-	int ret;
-
-	if (dt_open_and_check() < 0)
-		return -EINVAL;
-
-	if (fdt_get_address(&fdt) == 0)
-		return -EINVAL;
-
-	adc_node = fdt_node_offset_by_compatible(fdt, -1, "nxp,s32cc-adc");
-	if (adc_node == -1)
-		return -ENODEV;
-
-	ret = s32_adc_get_setup_from_fdt(fdt, adc_node, adc_driver);
-	if (ret)
-		return ret;
-
-	return s32_adc_init(adc_driver);
-}
-
-int ddr_config_fixup(void)
-{
-	struct s32_adc adc_driver;
-	uint32_t adc_val;
-	int rev_idx;
-	int ret;
-
-	ret = dt_init_adc(&adc_driver);
-	if (ret)
-		return ret;
-
-	ret = s32_adc_read_channel(&adc_driver, SARADC0_CH5, &adc_val);
-	if (ret)
-		return ret;
-
-	rev_idx =  find_rdb_rev(adc_val);
-	if (rev_idx < 0) {
-		ERROR("Unknown RDB board revision: ");
-		return 0;
-	}
-
-	switch (rdb_revisions[rev_idx].rev[0]) {
-	case 'C':
-	case 'D':
-	case 'E':
-		break;
-	case 'F':
-		return ddr_fixup_rdb_revf();
-	}
-
-	return 0;
-}
diff --git a/plat/nxp/s32/s32g/s32g_bl2_el3.c b/plat/nxp/s32/s32g/s32g_bl2_el3.c
index 9f021df91..ba615984c 100644
--- a/plat/nxp/s32/s32g/s32g_bl2_el3.c
+++ b/plat/nxp/s32/s32g/s32g_bl2_el3.c
@@ -25,13 +25,6 @@
 #include <dt-bindings/ddr-errata/s32-ddr-errata.h>
 #endif
 
-#pragma weak ddr_config_fixup
-
-int ddr_config_fixup(void)
-{
-	return 0;
-}
-
 static bl_mem_params_node_t s32g_bl2_mem_params_descs[6];
 REGISTER_BL_IMAGE_DESCS(s32g_bl2_mem_params_descs)
 
@@ -218,11 +211,6 @@ void bl2_el3_plat_arch_setup(void)
 
 	clear_swt_faults();
 
-	if (ddr_config_fixup()) {
-		ERROR("Failed to apply the DDR configuration fixup\n");
-		panic();
-	}
-
 	/* This will also populate CSR section from bl31ssram */
 	if (ddr_init()) {
 		ERROR("Failed to configure the DDR subsystem\n");
-- 
2.25.1

